File: TiaCodegen/Commands/Functions/Base/FunctionBlockCall.cs
using TiaCodegen.Interfaces;

namespace TiaCodegen.Commands.Functions.Base
{
    public class FunctionBlockCall : FunctionCall
    {
        public string InstanceName { get; set; }

        public FunctionBlockCall(string functionName, string instanceName, IOperationOrSignal eno = null) : base(functionName, eno)
        {
            FunctionName = functionName;
            InstanceName = instanceName;
        }
    }
}

File: TiaCodegen-ts/src/Commands/Functions/Base/FunctionBlockCall.ts
import { IOperationOrSignal } from '../../../Interfaces/IOperationOrSignal.js';
import { FunctionCall } from './FunctionCall.js';

export interface FunctionBlockCallOptions {
    functionName: string;
    instanceName: string;
    eno?: IOperationOrSignal | null;
}

export class FunctionBlockCall extends FunctionCall {
    instanceName: string;

    constructor(options: FunctionBlockCallOptions) {
        const { functionName, instanceName, eno = null } = options;
        super({ functionName, eno });
        this.instanceName = instanceName;
    }
}
